### ISSUE TITLE
Update HCV scoring rules

### ISSUE THREAD

**Issue by donkirkby** [cfe-lab/MiCall#436]
Anita is working on a new set of rules in a big spreadsheet.

* [x] parse 'resistance likely' and 'resistance possible' rules from the spreadsheet
* [x] avoid splitting within score terms
* [x] add reference sequence names
* [x] parse 'effect unknown' rules
* [x] ~~parse mutation combinations~~ Split to issue #440.
* [x] get codes and full names for new drugs
* [x] Allow clinical observations without fold shifts
* [x] Support < or </= in fold-shift data
* [x] overrides for GT2a/2b and GT4a/4d
* [x] wait for NS5b rules in regular tab format
* [x] ~~don't check wild type when subtype is overridden~~ (Postponed until subtypes are validated in #443.)
* [x] ~~always report subtype at top of report~~ (Postponed until subtypes are validated.)
* [x] ~~when subtype is not "a", include note on report: "resistance inferred from subtype Xa"~~ (Postponed until subtypes are validated.)
* [x] ~~special case to display note when a mutation only has data from 2b or 4d~~ (Postponed until subtypes are validated.)
* [x] special case for Sofosbuvir in genotype 6? different rules for Harvoni and Epclusa? should the 6e special case be on the other Sofosbuvir?
* [x] special case for Sofosbuvir: all mutations at position 282 are resistance likely (don't check fold shift)
* [x] combine GT2 back into single tab and same for GT4
* [x] ~~mark mutations with an asterisk when the variant matches the wild type, because it's from a different subtype~~ (Postponed until subtypes are validated.)
* [x] remove Boceprevir and Telaprevir from the reports, because they're obsolete
* [x] look for multiple mutations with the same variant in the same genotype, but different scores

**Comment by jeff-k** [comment]
For parsing spreadsheet we want to check if there are any other combination/individual variants that are a proper subset of the variants in another combination. If a subset of a combination's variants score higher than that combination, then that subset will also be called and the mitigating component will not be considered.

So, if (Q80K AND V168Y) => 8 but (Q80K AND V168Y AND L82P) => 4, where L82P is the mitigating variant, we need to make sure we take the minimum of these two scores and evaluate them only once.

When we process the spreadsheet we should check if any other rules are entailed by the one we're considering and construct a MIN ( ... ) expression with them both.

Attached is a comma separated list of rules that satisfies the expected scores in the spreadsheet.
[hcv_rules.txt](https://github.com/cfe-lab/MiCall/files/1798747/hcv_rules.txt)

**Comment by jeff-k** [comment]
Here's the rules with disjoint MAX combinations split up like MINS are and some more redundant rules eliminated:
[hcv_rules_split_max.txt](https://github.com/cfe-lab/MiCall/files/1798884/hcv_rules_split_max.txt)

**Comment by donkirkby** [comment]
I linked commits a7bff811dec84649bfa7a7a63ac4cdbc755d669a and cbdb221d8a6571c802bbae78f8e76708293798a6 to issue #440 when I should have linked them to this issue.
